decode_value.go

```diff
@@ -204,6 +204,21 @@ func decodeInterfaceValue(d *Decoder, v reflect.Value) error {
 	}
 
 	elem := v.Elem()
+
+	//support nested interface{}
+	elemTyp := elem.Type()
+	if elemTyp.Implements(customDecoderType) {
+		return decodeCustomValue(d, v)
+	}
+
+	if elemTyp.Implements(unmarshalerType) {
+		return unmarshalValue(d, v)
+	}
+
+	if elemTyp.Implements(binaryUnmarshalerType) {
+		return unmarshalBinaryValue(d, v)
+	}
```

Comment:
Presumably this should not be needed because it should be handled by DecodeValue. Do you have an idea why it does not work?

Reply:
https://github.com/vmihailenco/msgpack/issues/272 mentioned， we often need to dynamically get the value from interface according to different type， but v5 only used like:
`
var test = &msgpack.RawMessage
msgpack.Unmarshal(data, test)

`

in my practice, use msgpack serialize the rpc response struct{Data interface{}}, and need to deserialize Data(like struct not map[string]inerface{}) according to different types,  use json.rawmessage can work, unserialize  Data as rawmessage  and both get other field, but msgpack not support 
 json example:
`
type Test struct {
                 Action int
		Data interface{}
}

var test = &Test{
		Data: &json.RawMessage{},
	}
	if err := json.Unmarshal(data, &test); err != nil {
		logger.Fatalf("unexpected error:%s", err.Error())
	}

        switch test.Action {
           case ActionRes:
	      var res = &Res{}
	     if err :=json.Unmarshal(*test2.Arg2.(*json.RawMessage), res); err != nil {
                  logger.Errorf("%s", err.Error())
             }
         case ActionInt:
               ......
       }
`